### ISSUE TITLE
Managing a copied `node_modules` is tricky

### ISSUE THREAD

**Issue by dhess** [tweag/npmlock2nix#86]
**Is your feature request related to a problem? Please describe.**

Unfortunately, it's necessary to use `node_modules_mode = "copy"` with `create-react-app` because it wants to write `node_modules/.cache`. (I've asked upstream for the ability to change this location: see https://github.com/facebook/create-react-app/issues/11263)

Using this mode is tricky to get right. So far I've discovered the following:

* You probably want to use `--package-lock-only` with `npm install` and similar commands.
* You might have to `rm -rf node_modules` before running any `npm install` commands. Otherwise, it's possible to get Nix store links in your `package-lock.json` file, which is probably not what you want. See https://github.com/hackworthltd/react-nix-craco-public/commit/37b852373a507fe5b7419c1ce2c92cbae14f5593 and #85.

**Describe the solution you'd like**

The project could benefit from some documentation along these lines. It might also be helpful if the shell included some `npm` wrapper scripts that enforce these best practices. Adding an optional `shellHook` to remove the `node_modules` directory when you're in `copy` mode would be helpful.

**Describe alternatives you've considered**

We've written some scripts to enforce these best practices in our project, but it's still easy to get something wrong, and the learning curve is a bit steep if you're not familiar with `npm`.

**Comment by RaitoBezarius** [comment]
As I have been bitten by all these things, create-react-app writing in a cache folder, moving to Vite.js to avoid this, managing a copied `node_modules` through Nix, I am interested into writing the docs.

Plus, I am interested into fixing the root cause of #85 as it is seriously a pain for the developer experience :/.

**Comment by dhess** [comment]
Yes, we also switched to Vite from CRA, but additionally, we've also had to switch from `npmlock2nix` to `yarn2nix` for numerous other reasons, most of which are filed as issues in this project's tracker.

I think `npmlock2nix` has the most potential of all of the current Nix+Node.js integrations, but unfortunately, the number of outstanding issues and the apparent lack of an active maintainer mean that I don't think we'll be able to use it anytime soon. I'm not faulting anybody, by the way. I'm glad this project exists and hope that someone picks it up again someday.

**Comment by RaitoBezarius** [comment]
@dhess Curious on how you used `yarn2nix` as I switched from that initially because it has no support for copied `node_modules` and has some shortcomings in regards to easy production bundle / running tests / etc, nevertheless, thank you for your experience report :)

**Comment by dhess** [comment]
@RaitoBezarius It has required some bespoke `buildPhase` overrides. I'm happy to share, but this is not the forum for it. Feel free to contact me separately.

**Comment by adisbladis** [comment]
Over in https://github.com/adisbladis/buildNodeModules I've taken another approach:
I let `node_modules` be a directory and link individual packages into it.

To not overwrite files accidentally I took some inspiration from home-manager:
I only consider a file/directory "managed" if it's a symlink to the Nix store.
